### PR TITLE
fixed unknown block or directive error

### PR DESCRIPTION
**Pull request by mofantor**
* Added WithSkipValidBlocks option to skip valid sub-directives in blocks
* Skip valid blocks by default (e.g., types, map)

### CODE REVIEW

File: parser/parser.go [hunk 1]
```diff
@@ -194,7 +204,7 @@ func (p *Parser) Parse() (*gonginx.Config, error) {
 }
 
 // ParseBlock parse a block statement
-func (p *Parser) parseBlock(inBlock bool) (*gonginx.Block, error) {
+func (p *Parser) parseBlock(inBlock bool, isSkipValidDirective bool) (*gonginx.Block, error) {
```
Comment (tufanbarisyildirim) [inline parser/parser.go:207]: @mofantor  I like the solution here, do you think you can pass the all options (for the future when we need to check more options recursively) here?

Comment (mofantor) [reply parser/parser.go:207]: probably not, and if we need to check more options ,we should use options collection to pass and validate( for future)

Comment (mofantor) [reply parser/parser.go:207]: @tufanbarisyildirim I have just finished two tasks in the project and I am ready to submit PR. If you are interested, you can come to the project and have a look。

- move any context wrapper into their own file (remove from parser)
- link the parent directive to any directive for easier manipulation

Comment (mofantor) [reply parser/parser.go:207]: @tufanbarisyildirim project : https://github.com/mofantor/gonginx.git

Comment (tufanbarisyildirim) [reply parser/parser.go:207]: Yeah I already saw them, would love to see they are merging here! I will allocate some more time to go over today.

File: parser/skip_valid_block.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package parser
+
+import "strings"
+
+var skipValidBlocks = `types
+map
+`
+
+var SkipValidBlocks map[string]struct{} = map[string]struct{}{}
```
Comment (tufanbarisyildirim) [inline parser/skip_valid_block.go:9]: can you write a small comment here to make the linter happy? @mofantor

**Review by tufanbarisyildirim — CHANGES_REQUESTED** [review]

File: parser/skip_valid_block.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package parser
+
+import "strings"
+
```
Comment (mofantor) [inline parser/skip_valid_block.go:4]: // skipValidBlocks defines a list of valid blocks to be skipped during initialization.
// This string is split by newline characters, with each trimmed block name added to the SkipValidBlocks mapping.

File: parser/skip_valid_block.go (hunk 2 above)
Comment (mofantor) [inline parser/skip_valid_block.go:9]: // SkipValidBlocks is a mapping used to store names of valid blocks that should be skipped.
// An empty struct is used as the value type to occupy minimal space, serving only as a presence indicator

**Pull request merged**